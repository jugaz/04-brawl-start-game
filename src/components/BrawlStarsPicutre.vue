<template>
   <div class="brawlstars-container">
      <img 
         v-if="!showBrawlStart"
         :src="imgSrc"
         class="hidden-brawlstars"
         alt="brawlstars">
 
      <img v-else class="fade-in" :src="imgSrc" alt="brawlstars">
   </div>
   
</template>

<script>
export default {
   props: {
      brawlStartId: {
         type: Number,
         require: true
      },
      showBrawlStart: {
         type: Boolean,
         require: true,
         default: false

      }
   },
   computed: {
      imgSrc() {
          return `https://jugaz.github.io/brawl-start-api/images/${this.brawlStartId}.png`
      }
   }

}
</script>

<style scoped>

   .brawlstars-container {
      height: 300px;
   }
   img {
      height: 300px;

      user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;
      -webkit-user-drag: none;
      -webkit-user-select: none;
   }
   .hidden-brawlstars {
      filter: brightness(0);
   }
</style>