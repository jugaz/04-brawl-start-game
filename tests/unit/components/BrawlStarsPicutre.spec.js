import { shallowMount } from "@vue/test-utils";
import BrawlStarsPicutre from "@/components/BrawlStarsPicutre";

describe("BrawlStarsPicutre component", () => {
  test("debe de hacer match con el snapshot", () => {
    const wrapper = shallowMount(BrawlStarsPicutre, {
      props: {
        brawlStartId: 1,
        showBrawlStart: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe mostrar la imagen oculta y el brawlstart 50", () => {
    const wrapper = shallowMount(BrawlStarsPicutre, {
      props: {
        brawlStartId: 50,
        showBrawlStart: false,
      },
    });

    const [ img1, img2 ] = wrapper.findAll("img")


    expect(img1.exists()).toBeTruthy();
    expect(img2).toBe(undefined);
    
    expect( img1.classes('hidden-brawlstars') ).toBeTruthy()
    expect( img1.attributes ).toBeTruthy()

    expect( img1.attributes("src") ).toBe('https://jugaz.github.io/brawl-start-api/images/50.png')
    
  });

  test("debe de mostar el brawlstart si el showBrawlStart:true", () => {

    const wrapper = shallowMount(BrawlStarsPicutre, {
        props: {
          brawlStartId: 50,
          showBrawlStart: true,
        },
      });
  
      const img1 = wrapper.find("img");
  
      expect(img1.exists()).toBeTruthy();
      
      expect( img1.classes('hidden-brawlstars') ).toBe(false);
      expect( img1.classes('fade-in') ).toBe(true);
  });
});
