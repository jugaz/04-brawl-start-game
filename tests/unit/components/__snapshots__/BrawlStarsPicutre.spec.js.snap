// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrawlStarsPicutre component debe de hacer match con el snapshot 1`] = `<div class="brawlstars-container"><img src="https://jugaz.github.io/brawl-start-api/images/1.png" class="hidden-brawlstars" alt="brawlstars"></div>`;
